packages/x-date-pickers-pro/src/DateRangePickerDay/DateRangePickerDay.tsx

```diff
@@ -228,8 +228,10 @@ const DateRangePickerDayDay = styled(PickersDay, {
   }),
   ...(!ownerState.isHighlighting &&
     !ownerState.isDragging && {
-      '&:hover': {
-        border: `1px solid ${(theme.vars || theme).palette.grey[500]}`,
+      '@media (pointer: fine)': {
```

Comment:
Found this with as a solution to avoid unwanted hover styles on mobile being stuck on previously clicked date (where touch did not start).